### PR TITLE
Tech Debt : Display proper error message when update pin map step is…

### PR DESCRIPTION
**Pull request by prat-mahendran**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- We use the `self._sequence_context.Execution.RunTimeVariables.Exists` method call to determine if the field `PinMapId` exists in TestStand. It returns a Boolean value. If the field is not available, we throw a runtime error; if it is available, we proceed and return its value.
- 
### Why should this Pull Request be merged?

- This presents a user-friendly error message when a user overlooks adding an "Update Pin Map" step to their sequencer, ensuring easy understanding of the issue.

### Before

before_python

### After

after

### What testing has been done?

Manually verified the example execution.

### CODE REVIEW

**Review by jayaseelan-james — APPROVED** [review]
Please ensure to update the changes in all of its occurrences and update the PR title and description respectively.

File: examples/game_of_life/_helpers.py [hunk 1]
```diff
@@ -26,9 +26,12 @@ def get_active_pin_map_id(self) -> str:
         Returns:
             The resource id of the pin map that is registered to the pin map service.
         """
-        return self._sequence_context.Execution.RunTimeVariables.GetValString(
-            "NI.MeasurementLink.PinMapId", 0x0
-        )
+        execution = self._sequence_context.Execution.RunTimeVariables
+        if not execution.Exists("NI.MeasurementLink.PinMapId", 0x0):
+            raise RuntimeError(
+                "Failed to retrieve the active pin map ID .Please check the Update Pin Map step ."
```
Comment (SowmyaRavee) [inline examples/game_of_life/_helpers.py:32]: Please correct the space between the word and punctuation

Comment (jayaseelan-james) [reply examples/game_of_life/_helpers.py:32]: You might want to take a look at this [comment](https://github.com/ni/measurementlink-python/pull/719#discussion_r1595235565) :)

[the remark this answers is not in the document]
Comment (SowmyaRavee) [reply examples/game_of_life/_helpers.py:32]: We don't have to move the Possible reason to new line
`"Failed to retrieve the registered pin map ID. Possible reason: The sequence might not include an 'Update Pin Map' step."`

File: examples/game_of_life/_helpers.py [hunk 2]
```diff
@@ -26,9 +26,12 @@ def get_active_pin_map_id(self) -> str:
         Returns:
             The resource id of the pin map that is registered to the pin map service.
         """
-        return self._sequence_context.Execution.RunTimeVariables.GetValString(
-            "NI.MeasurementLink.PinMapId", 0x0
-        )
+        execution = self._sequence_context.Execution.RunTimeVariables
```
[the remark this answers is not in the document]
Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: Variable name changed .

File: examples/game_of_life/_helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: Error message modified.

Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: Noted.

File: examples/game_of_life/_helpers.py [hunk 3]
```diff
@@ -26,9 +26,12 @@ def get_active_pin_map_id(self) -> str:
...
+        if not execution.Exists("NI.MeasurementLink.PinMapId", 0x0):
+            raise RuntimeError(
+                "Failed to retrieve the active pin map ID .Please check the Update Pin Map step ."
+            )
+        return execution.GetValString("NI.MeasurementLink.PinMapId", 0x0)
```
[the remark this answers is not in the document]
Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: Added.

File: examples/game_of_life/_helpers.py [hunk 4]
```diff
@@ -26,9 +26,13 @@ def get_active_pin_map_id(self) -> str:
         Returns:
             The resource id of the pin map that is registered to the pin map service.
         """
-        return self._sequence_context.Execution.RunTimeVariables.GetValString(
-            "NI.MeasurementLink.PinMapId", 0x0
-        )
+        PIN_MAP_ID = "NI.MeasurementLink.PinMapId"
```
Comment (dixonjoel) [inline examples/game_of_life/_helpers.py:29]: I think this should be a constant at the top level of the file and internal like:
```
_PIN_MAP_ID = "NI.MeasurementLink.PinMapId"
```

Comment (dixonjoel) [reply examples/game_of_life/_helpers.py:29]: @bkeryan ?

**Review by dixonjoel — APPROVED** [review]
Approved with suggestions once the checks are all passing.

File: examples/game_of_life/_helpers.py (hunk 4 above)
Comment (bkeryan) [reply examples/game_of_life/_helpers.py:29]: I agree. Also rename to indicate that this is a runtime var name, not the actual id

**Review by SowmyaRavee — APPROVED** [review]
Please fix the lint error and update the variable name

File: examples/game_of_life/_helpers.py [hunk 5]
```diff
@@ -26,9 +26,13 @@ def get_active_pin_map_id(self) -> str:
         Returns:
             The resource id of the pin map that is registered to the pin map service.
         """
-        return self._sequence_context.Execution.RunTimeVariables.GetValString(
-            "NI.MeasurementLink.PinMapId", 0x0
-        )
+        pin_map_id = "NI.MeasurementLink.PinMapId"
```
Comment (bkeryan) [inline examples/game_of_life/_helpers.py:29]: > [bkeryan](https://github.com/bkeryan) [19 hours ago](https://github.com/ni/measurementlink-python/pull/719#discussion_r1595819279) • 
> I agree. Also rename to indicate that this is a run time variable name, not the actual id

`pin_map_id` sounds like the id itself. Call the variable or constant something like `pin_map_id_var` to show that it's the variable name.

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: examples/game_of_life/_helpers.py (hunk 5 above)
Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: I have changed the variables names and made it a top level constant as well.

File: examples/game_of_life/_helpers.py (hunk 4 above)
Comment (prat-mahendran) [reply examples/game_of_life/_helpers.py]: Modified.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**